### ISSUE TITLE
`require-explicit-slots` doesn't recognize interface passed to the macro

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#2398]
**Tell us about your environment**

- **ESLint version:** `8.56.0`
- **eslint-plugin-vue version:** `9.21.1`
- **Vue version:** `3.4.15`
- **Node version:** `20.10.0`
- **Operating System:** `Mac OS 14.2.1`

**Please show your full configuration:**

```json5
module.exports = {
  parser: 'vue-eslint-parser',

  parserOptions: {
    parser: '@typescript-eslint/parser',
  },

  extends: ['plugin:vue/base'],

  rules: {
    'vue/require-explicit-slots': 'error',
  },
};
```

**What did you do?**	

```vue
<template>
  <div>
    <slot />
  </div>
</template>

<script lang="ts" setup>
interface Slots {
  default: () => string;
}

defineSlots<Slots>();
</script>

```

**What did you expect to happen?**
No error reported.

**What actually happened?**
```
test.vue
  3:5  error  Slots must be explicitly defined  vue/require-explicit-slots

✖ 1 problem (1 error, 0 warnings)
```

**Repository to reproduce this issue**
https://stackblitz.com/edit/stackblitz-starters-reg1dy?file=test.vue

```
~$ npm run lint
```

**Comment by Ky6uk** [comment]
Updated packages to the latest version.

**Comment by daniluk4000** [comment]
Same with this generic

![image](https://github.com/vuejs/eslint-plugin-vue/assets/10450717/8bd157f7-8a12-450a-bdc5-4689391947c6)

**Comment by sanderdewinter** [comment]
I just opened #2484, feel free to contribute
